src/v/coproc/script_context.cc

```diff
@@ -174,14 +175,13 @@ ss::future<> script_context::process_one_reply(process_batch_reply::data e) {
         return ss::now();
     }
     auto ntp_ctx = found->second;
-    return write_materialized(materialized_ntp, std::move(*e.reader))
-      .then([this, ntp_ctx](write_response wr) {
-          if (wr == write_response::crc_failure) {
+    model::term_id highest_term = ntp_ctx->log.offsets().dirty_offset_term;
```

Comment:
why batches are reordered ? I think I do not fully understand this

Reply:
The API to a coprocessor is 
```
ss::future<map<topic, record_batch_reader>> apply(topic, record_batch_reader)
```
the issue observed is what would happen if the records in the reader were reordered (totally fine) but if we just blindly wrote those records to storage?

If the term ids were 1-2-3-3-3-4 and the user reordered them in such a way they no longer are monotonically increasing , then storage will throw. So the solution that was decided was just to rewrite term ids before writing these records to storage.

Reply:
To clairfy there isn't a 1-1 representation between a source topic and materialized. The coprocessor may filter, or inject more records